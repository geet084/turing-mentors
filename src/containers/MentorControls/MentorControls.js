import React, { Component } from 'react';

export class MentorControls extends Component {
  render() {
    const { handleChange, javascript, ruby, denver, remote } = this.props;
  
    return (
      <div>
        <input type="checkbox" onClick={handleChange} name="javascript" />JS
        <input type="checkbox" onClick={handleChange} name="ruby" />RUBY
        <input type="checkbox" onClick={handleChange} name="denver" />Denver
        <input type="checkbox" onClick={handleChange} name="remote" />Remote
        <div>
          {javascript && <span>{javascript}</span>}
          {ruby && <span>{ruby}</span>}
          {denver && <span>{denver}</span>}
          {remote && <span>{remote}</span>}
        </div>
      </div>
    )
  }
}

export default MentorControls;