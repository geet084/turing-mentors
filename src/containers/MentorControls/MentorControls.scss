// .mentor-controls {
//   min-width: 430px;
//   margin-left: -30px;
//   transform: scale(0.70);
//   text-decoration: none;
//   color: grey;
//   font-size: 2.25rem;
//   padding: 2px 4px;
//   border: 1px solid white;
//   border-radius: 10px;
//   box-shadow: 4px 4px 8px -2px rgba(0,0,0,0.25);
//   &:hover {
//     cursor: pointer;
//     margin-bottom: -2px;
//     border-bottom: 2px solid #12f1fc;
//   }
// }