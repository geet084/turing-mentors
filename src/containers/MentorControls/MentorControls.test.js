import React from 'react';
import { shallow } from 'enzyme';
import { MentorControls } from './MentorControls';

describe('MentorControls', () => {
  let wrapper;
  const mockProps = {
    handleChange: jest.fn(),
    javascript: false,
    ruby: false,
    denver: false,
    remote: false,
  }

  beforeEach(() => {
    wrapper = shallow(<MentorControls {...mockProps} />);
  });

  describe('initial', () => {
    it('should have match the correct snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});