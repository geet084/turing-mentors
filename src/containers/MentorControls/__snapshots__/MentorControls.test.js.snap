// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MentorControls initial should have match the correct snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MentorControls
    denver={false}
    handleChange={[MockFunction]}
    javascript={false}
    remote={false}
    ruby={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          name="javascript"
          onClick={[MockFunction]}
          type="checkbox"
        />,
        "JS",
        <input
          name="ruby"
          onClick={[MockFunction]}
          type="checkbox"
        />,
        "RUBY",
        <input
          name="denver"
          onClick={[MockFunction]}
          type="checkbox"
        />,
        "Denver",
        <input
          name="remote"
          onClick={[MockFunction]}
          type="checkbox"
        />,
        "Remote",
        <div />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "javascript",
          "onClick": [MockFunction],
          "type": "checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "JS",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "ruby",
          "onClick": [MockFunction],
          "type": "checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "RUBY",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "denver",
          "onClick": [MockFunction],
          "type": "checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "Denver",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "remote",
          "onClick": [MockFunction],
          "type": "checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "Remote",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            false,
            false,
            false,
          ],
        },
        "ref": null,
        "rendered": Array [
          false,
          false,
          false,
          false,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="javascript"
            onClick={[MockFunction]}
            type="checkbox"
          />,
          "JS",
          <input
            name="ruby"
            onClick={[MockFunction]}
            type="checkbox"
          />,
          "RUBY",
          <input
            name="denver"
            onClick={[MockFunction]}
            type="checkbox"
          />,
          "Denver",
          <input
            name="remote"
            onClick={[MockFunction]}
            type="checkbox"
          />,
          "Remote",
          <div />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "javascript",
            "onClick": [MockFunction],
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "JS",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "ruby",
            "onClick": [MockFunction],
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "RUBY",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "denver",
            "onClick": [MockFunction],
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "Denver",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "remote",
            "onClick": [MockFunction],
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "Remote",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              false,
              false,
              false,
            ],
          },
          "ref": null,
          "rendered": Array [
            false,
            false,
            false,
            false,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
